libpromises/cf3parse_logic.h

```diff
@@ -895,8 +899,19 @@ static inline void ParserHandleBlockAttributeRval()
         const BodySyntax *body_syntax = BodySyntaxGet(P.block, P.blocktype);
         assert(body_syntax != NULL);
 
-        const ConstraintSyntax *constraint_syntax =
-            BodySyntaxGetConstraintSyntax(body_syntax->constraints, P.lval);
+        ConstraintSyntax *constraint_syntax = NULL;
```

Comment:
Since all branches assign to constraint syntax, it would be more appropriate to leave it uninitialized here.

```suggestion
        ConstraintSyntax *constraint_syntax;
```